Add docs for AWS Cloudmap and Nameformat name resolution components (#4694)
dapr/docs: daprdocs/content/en/reference/components-reference/supported-name-resolution/nr-nameformat.md

---
type: docs
title: "Nameformat"
linkTitle: "NameFormat"
description: Detailed information on the NameFormat name resolution component
---

> The Name Format name resolver provides a flexible way to resolve service names using a configurable format string with placeholders. This is useful in scenarios where you want to map service names to predictable DNS names following a specific pattern.

Reviewer: It would be good to provide some more details on example use cases for this component, more of the "why" behind using this and when. This is ok, but it is light on scenarios, so let's include a few.

Author: Updated

## Configuration Format

> 

Reviewer: Name resolution is configured via the [Dapr Configuration]({{< ref configuration-overview.md >}}).

Within the configuration YAML, set the `spec.nameResolution.component` property to `"nameformat"`, then pass configuration options in the `spec.nameResolution.configuration` dictionary.

Author: Updated

To use the Name Format name resolver, create a configuration in your Dapr environment:

```yaml
apiVersion: dapr.io/v1alpha1
kind: Configuration
metadata:
  name: appconfig
spec:
  nameResolution:
    component: "nameformat"
    configuration:
      format: "service-{appid}.default.svc.cluster.local"  # Replace with your desired format pattern
```

> ## Configuration Fields

Reviewer: ```suggestion
## Spec configuration fields
```

| Field   | Required | Details | Example |
|---------|----------|---------|---------|

> | format  | Y | The format string to use for name resolution. Must contain the `{appid}` placeholder which will be replaced with the actual service name. | `"service-{appid}.default.svc.cluster.local"` |

Reviewer: ```suggestion
| format  | Y | The format string to use for name resolution. Must contain the `{appid}` placeholder which is replaced with the actual service name. | `"service-{appid}.default.svc.cluster.local"` |
```

## Examples

> When configured with `format: "service-{appid}.default.svc.cluster.local"`, the resolver will transform service names as follows:

Reviewer: ```suggestion
When configured with `format: "service-{appid}.default.svc.cluster.local"`, the resolver transforms service names as follows:
```

- Service ID "myapp" → "service-myapp.default.svc.cluster.local"
- Service ID "frontend" → "service-frontend.default.svc.cluster.local"

## Notes

> - Empty service IDs are not allowed and will result in an error

Reviewer: ```suggestion
- Empty service IDs are not allowed and results in an error.
```
